src/WebJobs.Script/Binding/ExtensionBinding.cs

```diff
@@ -70,6 +71,12 @@ public override Collection<CustomAttributeBuilder> GetCustomAttributes(Type para
                     context.Value = result;
                 }
             }
+            else if (_binding.DefaultType == typeof(JArray))
+            {
+                JArray entityArray = await context.Binder.BindAsync<JArray>(_attributes.ToArray());
+                string json = entityArray.ToString(Formatting.None);
```

Comment:
Why is this conversion needed? I think later stages in the pipeline should convert as necessary. Note that we don't convert JObjects to string above.